docs/spec/spec.md

```diff
@@ -0,0 +1,173 @@
+# Specification: Ballerina LDAP Library
+
+_Authors_: @Nuvindu \
+_Reviewers_:  \
+_Created_: 2024/08/11 \
+_Updated_: 2024/08/11 \
+_Edition_: Swan Lake
+
+## Introduction
+
+LDAP (Lightweight Directory Access Protocol) is a vendor-neutral software protocol for accessing and maintaining distributed directory information services. It allows users to locate organizations, individuals, and other resources such as files and devices in a network. LDAP is used in various applications for directory-based authentication and authorization.
+
+The Ballerina LDAP module provides the capability to efficiently connect, authenticate, and interact with directory servers. It allows users to perform operations such as searching for entries, and modifying entries in an LDAP directory, providing better support for directory-based operations.
+
+If you have any feedback or suggestions about the library, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues) or in the [Discord server](https://discord.gg/ballerinalang). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
```

Comment:
Spec Introduction follows a standard template [1] Any reason to move away from that ?

[1] - https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/docs/spec/spec.md#introduction